### PR TITLE
Add distribution unit checks for --force-full on sync

### PR DESCRIPTION
**Pull request by bherrin3**
Pulp 2.19 fixes a bug where distribution units for kickstart
files were not synched on a --force-full.

Refactored ForceSyncTestCase for distribution units and any future
type of unit to be tested with force-full with resync.

Added a test case  to support checks on distribution units and
make the test_sync.py test more extensible for other units using
a tuple definition of the location and extension of the unit.

The rpm-kickstart is added to the fixtures for test. The fixture
was derived from the Fedora 29 ISO and scaled down for test.

See: https://pulp.plan.io/issues/4267

closes #4409

### CODE REVIEW

File: pulp_2_tests/constants.py [hunk 1]
```diff
@@ -520,6 +520,9 @@
 RPM_INVALID_URL = urljoin(RPM_INVALID_FEED_URL, 'invalid.rpm')
 """The URL to an invalid RPM package."""
 
+RPM_KICKSTART_FEED_URL = urljoin(PULP_FIXTURES_BASE_URL, 'rpm-kickstart/')
+"""The URL to a KICKSTART repository."""
+
```
Comment (kersommoura) [inline pulp_2_tests/constants.py]: What about add a note stating that this fixtures are not generated using pulp-fixtures? Something like...

```.rst
.. NOTE:: This repository is not generated by `pulp-fixtures`_.
```

File: pulp_2_tests/tests/rpm/cli/test_sync.py [hunk 2]
```diff
@@ -89,64 +90,117 @@ def test_all(self):
...
+        """Test a forced full sync on Pulp.
 
+        Using the tuple ``unit_type`` to target rpms units
+        and looking for the .rpm extension, respectively.
+        """
+        if not selectors.bug_is_fixed(1982, self.cfg.pulp_version):
+            self.skipTest('https://pulp.plan.io/issues/1982')
+        unit_type = ('rpm', 'rpm')
+        feed = RPM_UNSIGNED_FEED_URL
+        self._do_test(unit_type, feed)
```
Comment (kersommoura) [inline pulp_2_tests/tests/rpm/cli/test_sync.py]: What about we avoid to use those variables?

```suggestion
        self._do_test(('rpm', 'rpm'),  RPM_UNSIGNED_FEED_URL)
```

Comment (bherrin3) [reply pulp_2_tests/tests/rpm/cli/test_sync.py]: @kersommoura added in both locations, removed from docstrings,  rebuilt docs, re-tested and pushed.  Thank you for that optimization.

File: pulp_2_tests/constants.py [hunk 3]
```diff
@@ -520,6 +520,17 @@
 RPM_INVALID_URL = urljoin(RPM_INVALID_FEED_URL, 'invalid.rpm')
 """The URL to an invalid RPM package."""
 
+RPM_KICKSTART_FEED_URL = urljoin(PULP_FIXTURES_BASE_URL, 'rpm-kickstart/')
+"""The URL to a KICKSTART repository.
+
+NOTE:: This repository is not generated by `pulp-fixtures`_. Please see
```
Comment (kersommoura) [inline pulp_2_tests/constants.py]:
```suggestion
..NOTE:: This repository is not generated by `pulp-fixtures`_. Please see
```

See: https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#paragraph-level-markup

**Comment by bherrin3** [comment]
@kersommoura The test fixture has been updated. Added a `README` in the fixture explaining the differences from F29 for reference.

- https://repos.fedorapeople.org/pulp/pulp/fixtures/rpm-kickstart/

Now this only has the 6 .img files that would have been there, resized to 1MB each, and no longer has any conflicting RPMs to pollute the RPM content units test.

Re-ran looping full test through 10 iterations without failure, docs generated fine. Have not implemented `black` syntax on this PR. 

Should be ready for re-review.

**Pull request merged**